src/check-dsl.ts

```diff
@@ -62,7 +62,8 @@ const defaultError = (lines: string[]) => {
 // are tuple to user set.  If so, return the type and relationship.
 // Otherwise, return null as relationship
 function destructTupleToUserset(allowableType: string): string[] {
-  return allowableType.split("#", 2);
+  // For validation purpose, we don't care whether it is wildcard restricted or not
+  return allowableType.replace(":*", "").split("#", 2);
```

Comment:
We do care though. If it allows `*`, it should not be allowed on a TupleToUserset, same as the case with a relation.

In both cases, we won't expand, so we should warn them

Reply:
fixed